src/modules/grid.scss

```diff
@@ -1,51 +1,1289 @@
+        --grid-gutter: var(--grid-desktop-gutter);
+        --grid-columns: var(--grid-desktop-column);
+    }
+}
+
+@media (min-width: 1366px) {
+    :root {
+        --grid-margin: var(--grid-wide-margin);
+        --grid-gutter: var(--grid-wide-gutter);
+        --grid-columns: var(--grid-wide-column);
+    }
+}
+
+.grid {
+	display: grid;
+	grid-template-columns: repeat(var(--grid-columns), 1fr);
+	width: 100%;
+	column-gap: var(--grid-gutter);
+}
+
+.col-full {
+	grid-column: 1 / -1;
+}
+
+.col-1 {
+	grid-column: span 1;
+}
+
+.col-2 {
+	grid-column: span 2;
+}
+
+.col-3 {
+	grid-column: span 3;
+}
+
+.col-4,
+.col-5,
+.col-6,
+.col-7,
+.col-8,
+.col-9,
+.col-10,
+.col-11,
+.col-12 {
+	grid-column: span 4;
+}
+
+.col-start-1 {
+	grid-column-start: 1;
+}
+
+.col-start-2 {
+	grid-column-start: 2;
+}
+
+.col-start-3 {
+	grid-column-start: 3;
+}
+
+.col-start-4,
+.col-start-5,
+.col-start-6,
+.col-start-7,
+.col-start-8,
+.col-start-9,
+.col-start-10,
+.col-start-11,
+.col-start-12 {
+	grid-column-start: 4;
+}
+
+/*
+  .-col-start-1 is intentionally left out
+*/
+
+.-col-start-2 {
+	grid-column-start: -2;
+}
+
+.-col-start-3 {
+	grid-column-start: -3;
+}
+
+.-col-start-4 {
+	grid-column-start: -4;
+}
+
+.-col-start-5,
+.-col-start-6,
+.-col-start-7,
+.-col-start-8,
+.-col-start-9,
+.-col-start-10,
+.-col-start-11,
+.-col-start-12,
+.-col-start-13 {
+	grid-column-start: -5;
+}
+
+.col-end-last {
+	grid-column-end: -1;
+}
+
+/*
+  .col-end-1 is intentionally left out
+*/
+
+.col-end-2 {
+	grid-column-end: 2;
+}
+
+.col-end-3 {
+	grid-column-end: 3;
+}
+
+.col-end-4 {
+	grid-column-end: 4;
+}
+
+/*
+  This means "end this column where column 5 start",
+  which is the same as saying "end this column where column 4 end"
+*/
+.col-end-5,
+.col-end-6,
+.col-end-7,
+.col-end-8,
+.col-end-9,
+.col-end-10,
+.col-end-11,
+.col-end-12,
+.col-end-13 {
+	grid-column-end: 5;
+}
+
+.-col-end-1 {
+	grid-column-end: -1;
+}
+
+.-col-end-2 {
+	grid-column-end: -2;
+}
+
+.-col-end-3 {
+	grid-column-end: -3;
+}
+
+.-col-end-4,
+.-col-end-5,
+.-col-end-6,
+.-col-end-7,
+.-col-end-8,
+.-col-end-9,
+.-col-end-10,
+.-col-end-11,
+.-col-end-12 {
+	grid-column-end: -4;
+}
+
+/*
+	Classes used to define rows on the grid. To be used along the `.grid` class.
+	Rows height is defined by `--col-size`
+*/
+.grid-rows-1 {
+	grid-template-rows: var(--col-size);
+}
+
+.grid-rows-2 {
+	grid-template-rows: repeat(2, var(--col-size));
+}
+
+.grid-rows-3 {
+	grid-template-rows: repeat(3, var(--col-size));
+}
+
+.grid-rows-4 {
+	grid-template-rows: repeat(4, var(--col-size));
+}
+
+.grid-rows-5 {
+	grid-template-rows: repeat(5, var(--col-size));
+}
+
+.grid-rows-6 {
+	grid-template-rows: repeat(6, var(--col-size));
+}
+
+.row-full {
+	grid-row: 1 / -1;
+}
+
+.row-1 {
+	grid-row: span 1;
+}
+
+.row-2 {
+	grid-row: span 2;
+}
+
+.row-3 {
+	grid-row: span 3;
+}
+
+.row-4 {
+	grid-row: span 4;
+}
+
+.row-5 {
+	grid-row: span 5;
+}
+
+.row-6 {
```

Comment:
In altri framework dove si usa un'impostazione di questo tipo per grid (es. Tailwind) viene data anche la possibilità di parametrizzare il numero di row, forse per noi potrebbe non servire?